libdbus-sys/build_vendored.rs

```diff
@@ -0,0 +1,389 @@
+    //    not a bug
+    //  - -Wunused-parameter is annoying when writing callbacks that follow
+    //    a fixed signature but do not necessarily need all of its parameters
+    "-Wno-missing-field-initializers",
+    "-Wno-unused-parameter",
+    "-Wno-error=duplicated-branches",
+
+    // Under clang, need to ignore cast-align and sign-compare
+    // Reference: https://bugs.freedesktop.org/show_bug.cgi?id=10599#c5
+    "-Wno-error=cast-align",
+    "-Wno-error=sign-compare",
+
+    // Required for 1.14 but not 1.15+
+    "-Wno-error=unused-but-set-variable",
+
+    // General warnings for both C and C++
+    "-Warray-bounds",
+    "-Wchar-subscripts",
+    "-Wdouble-promotion",
+    "-Wduplicated-branches",
+    "-Wduplicated-cond",
+    "-Wfloat-equal",
+    "-Wformat-nonliteral",
+    "-Wformat-security",
+    "-Wformat=2",
+    "-Winit-self",
+    "-Winline",
+    "-Wlogical-op",
+    "-Wmissing-declarations",
+    "-Wmissing-format-attribute",
+    "-Wmissing-include-dirs",
+    "-Wmissing-noreturn",
+    "-Wnull-dereference",
+    "-Wpacked",
+    "-Wpointer-arith",
+    "-Wredundant-decls",
+    "-Wrestrict",
+    "-Wreturn-type",
+    "-Wshadow",
+    "-Wstrict-aliasing",
+    "-Wswitch-default",
+    "-Wswitch-enum",
+    "-Wundef",
+    "-Wunused-but-set-variable",
+    "-Wwrite-strings",
+
+    // Extra warnings just for C
+    "-Wdeclaration-after-statement",
+    "-Wimplicit-function-declaration",
+    "-Wjump-misses-init",
+    "-Wmissing-prototypes",
+    "-Wnested-externs",
+    "-Wold-style-definition",
+    "-Wpointer-sign",
+    "-Wstrict-prototypes",
+];
+
+/// Wrapper type for the libdbus config.
+///
+/// Boils down to a set of C defs/undefs.
+struct Config<'a> {
+    defs: HashMap<&'a str, Option<String>>,
+    undefs: Vec<&'a str>,
+}
+
+impl<'a> Config<'a> {
+    fn new() -> Self {
+        Self {
+            defs: HashMap::new(),
+            undefs: Vec::new(),
+        }
+    }
+
+    fn enable(&mut self, flag: &'a str) {
+        self.defs.insert(flag, None);
+    }
+
+    fn disable(&mut self, flag: &'a str) {
+        self.undefs.push(flag);
+    }
+
+    fn set<T: ToString>(&mut self, flag: &'a str, value: T) {
+        self.defs.insert(flag, Some(value.to_string()));
+    }
+
+    /// Iterator over all C defines, set with -Dopt=val
+    fn defs<'b>(&'b self) -> Iter<'b, &'a str, Option<String>> {
+        self.defs.iter()
+    }
+
+    /// Iterator over all undefs, that must be set with -Uopt
+    fn undefs<'b>(&'b self) -> std::slice::Iter<'b, &'a str> {
+        self.undefs.iter()
+    }
+}
+
+/// Config.h and dbus-arch-deps.h stubs (we generate config programatically
+/// instead to be passed directly to cc)
+fn generate_stubs() -> Result<(), Box<dyn Error>> {
+    let mut config = File::create("vendor/dbus/config.h")?;
+    let mut arch = File::create("vendor/dbus/dbus/dbus-arch-deps.h")?;
+
+    // Stub out the file.
+    config.write_all(b"
```

Comment:
Needs reference (if there is one?)

Reply:
Currently no reference. The actual config is created dynamically in `generate_config()` but `config.h` is required to exist since some `dbus` source files include it.